### ISSUE TITLE
logistic sigmoid function outputs NaN indeterminate on complex input

### ISSUE THREAD

**Issue by drewshark** [pytorch/pytorch#121588]
### 🐛 Describe the bug

When sending a complex input where the real part of each element is infinity/-infinity, sigmoid functions (`torch.special.explit` and `torch.nn.functional.sigmoid`) outputs NaN.
However, when I sent the first row of this complex input to sigmoid functions, they will return normal value.

Here is the code:

```
import numpy as np
import torch

input = torch.tensor([
    [-np.inf+0.j, np.inf+0.j, np.inf+0.j, np.inf+0.j],
    [np.inf+0.j, -np.inf+0.j, np.inf+0.j, -np.inf+0.j],
])
print(torch.special.expit(input))
# tensor([[nan+nanj, 1.-0.j, 1.-0.j, 1.-0.j],
#         [1.-0.j, nan+nanj, 1.-0.j, nan+nanj]])
        
print(torch.nn.functional.sigmoid(input))
# tensor([[nan+nanj, 1.-0.j, 1.-0.j, 1.-0.j],
#         [1.-0.j, nan+nanj, 1.-0.j, nan+nanj]])

print(torch.special.expit(input[0]))
# tensor([0.+0.j, 1.+0.j, 1.+0.j, 1.+0.j])

print(torch.nn.functional.sigmoid(input[0]))
# tensor([0.+0.j, 1.+0.j, 1.+0.j, 1.+0.j])

```

At least, sigmoid functions are element-wise operations and thus it may be weird that results on a specific element of the input tensor is affected by the total shape of the input tensor.
Moreover, please check if yielding NaN for a complex value whose real part is infinity is expected or not. 

### Versions

[pip3] numpy==1.25.2
[pip3] torch==2.1.0+cu121
[pip3] torchaudio==2.1.0+cu121
[pip3] torchdata==0.7.0
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.16.0
[pip3] torchvision==0.16.0+cu121
[pip3] triton==2.1.0
